### ISSUE TITLE
Create a NATS Jetstream Consumer programatically in the eventing-controller

### ISSUE THREAD

**Issue by nachtmaar** [kyma-project/kyma#15093]
**TL;DR**
Do not let the JetStream SDK create the NATS Subscription and Consumer, instead do it in **two separate steps**.

**Description**
As a first step in the implementation of https://github.com/kyma-project/kyma/issues/15087, the eventing team decided to **create the NATS Jetstream Consumer independent** of the NATS Subscription.
Currently in the Jetstream Reconciler, the NATS Subscription and the NATS Consumer are created [in one step](https://github.com/kyma-project/kyma/blob/317ed95ca93e83297bd4491f7e877838064b9fa8/components/eventing-controller/pkg/handlers/jetstream.go#L403).
The idea of this ticket is to create the consumer and the subscriptions in two steps and not let the Jetstream SDK handle this for us.

**Reasons**

The advantage of this is that when deleting a NATS Subscription with [jsSubscription.Unsubscribe()](https://pkg.go.dev/github.com/nats-io/nats.go#Subscription.Unsubscribe), the [NATS JS Consumer](https://docs.nats.io/nats-concepts/jetstream/consumers) will not be automatically get deleted. We do use the [JS Interest Policy](https://docs.nats.io/using-nats/developer/develop_jetstream/model_deep_dive), this means that the events will not be deleted as there is active `interest` in them.
This will allow us to **pause a NATS Subscription** without losing events (`jsSubscription.Unsubscribe()`).

**Attachments**

**Comment by nachtmaar** [comment]
Adding a consume should be possible with [js.jsCtx.AddConsumer()](https://github.com/nats-io/nats.go/blob/v1.16.0/jsm.go#L55)

**Comment by VladislavPaskar** [comment]
since that change was introduced in this [PR](https://github.com/kyma-project/kyma/pull/15762), we can close this issue.
